lib/api/CDataSummarizationJsonSerializer.cc

```diff
@@ -185,206 +218,181 @@ void CDataSummarizationJsonSerializer::addToJsonStream(TGenericLineWriter& write
+    TEncoderUPtr encodings;
+    std::stringstream jsonStrm;
+    rapidjson::OStreamWrapper wrapper{jsonStrm};
+    CDataSummarizationJsonWriter::TGenericLineWriter writer{wrapper};
+    writer.StartObject();
+    writer.Key(JSON_ENCODINGS_TAG);
+    writer.write(doc[JSON_ENCODINGS_TAG]);
+    writer.EndObject();
+    core::CJsonStateRestoreTraverser traverser{jsonStrm};
+    encodings = std::make_unique<maths::CDataFrameCategoryEncoder>(traverser);
+
+    frame.columnNames(columnNames);
+    frame.categoricalColumns(columnIsCategorical);
+    frame.categoricalColumnValues(categoricalColumnValues);
+
+    TStrVec rowVec;
+    rowVec.reserve(numberColumns);
+    for (const auto& row : ifExists(JSON_DATA_TAG, getArray, doc)) {
+        for (const auto& column : getArray(row)) {
+            rowVec.emplace_back(getString(column));
         }
-    } else {
-        LOG_ERROR(<< "Data summarization field '" << JSON_CATEGORICAL_COLUMN_VALUES_TAG
-                  << "' is missing or has an unexpected format.");
-        return nullptr;
+        frame.parseAndWriteRow(
+            core::CVectorRange<const TStrVec>(rowVec, 0, rowVec.size()));
+        rowVec.clear();
     }
+    frame.finishWritingRows();
 
-    if (doc.HasMember(JSON_ENCODINGS_TAG) && doc[JSON_ENCODINGS_TAG].IsObject()) {
-        std::stringstream jsonStrm;
-        rapidjson::OStreamWrapper wrapper{jsonStrm};
-        CDataSummarizationJsonSerializer::TGenericLineWriter writer{wrapper};
-        writer.StartObject();
-        writer.Key(JSON_ENCODINGS_TAG);
-        writer.write(doc[JSON_ENCODINGS_TAG].GetObject());
-        writer.EndObject();
-        core::CJsonStateRestoreTraverser traverser{jsonStrm};
-        encoder = std::make_unique<maths::CDataFrameCategoryEncoder>(traverser);
-    } else {
-        LOG_ERROR(<< "Data summarization field '" << JSON_ENCODINGS_TAG
-                  << "' is missing or has an unexpected format.");
+    return encodings;
+}
+
+CRetrainableModelJsonReader::TEncoderUPtr
+CRetrainableModelJsonReader::dataSummarizationFromDocumentCompressed(TIStreamSPtr istream,
+                                                                     core::CDataFrame& frame) {
+    if (istream == nullptr) {
         return nullptr;
     }
+    try {
+        rapidjson::IStreamWrapper isw{*istream};
+        rapidjson::Document doc;
+        doc.ParseStream(isw);
+        auto compressedDataSummarization =
+            ifExists(JSON_COMPRESSED_DATA_SUMMARIZATION_TAG, getObject, doc);
+        TStreamBuffer buffer{ifExists(JSON_DATA_SUMMARIZATION_TAG, getString,
+                                      compressedDataSummarization),
+                             ifExists(JSON_DATA_SUMMARIZATION_TAG, getStringLength,
+                                      compressedDataSummarization)};
+        return dataSummarizationFromJsonStream(decompressStream(buffer), frame);
+    } catch (...) {
```

Comment:
Same as previous comment regarding eating low level errors.

Reply:
Done in [f2c3588](https://github.com/elastic/ml-cpp/pull/1894/commits/f2c35881e9ec52f95d64cd4ba16bc45c3712fed3).